src/parquet/reader-test.cc

```diff
@@ -149,13 +150,43 @@ TEST_F(TestAllTypesPlain, ColumnSelectionOutOfRange) {
 
   std::list<int> columns;
   columns.push_back(100);
-  ASSERT_THROW(reader_->DebugPrint(ss, columns), ParquetException);
+  ASSERT_THROW(reader_->DebugPrint(ss, columns, 128), ParquetException);
 
   columns.clear();
   columns.push_back(-1);
-  ASSERT_THROW(reader_->DebugPrint(ss, columns), ParquetException);
+  ASSERT_THROW(reader_->DebugPrint(ss, columns, 128), ParquetException);
 }
 
+TEST_F(TestAllTypesPlain, MemoryEstimation) {
+  std::stringstream ss;
+
+  int64_t batch_size = 128;
+  std::list<int> columns;
+
+  columns.push_back(0);
+  columns.push_back(5);
+  ParquetFileReader::MemoryUsage memory_usage1 =
+      reader_->EstimateMemoryUsage(false, columns, batch_size);
+  reader_->DebugPrint(ss, columns, batch_size, true);
+  ASSERT_GT(allocator_.MaxMemory(), memory_usage1.memory);
+  ASSERT_TRUE(memory_usage1.has_dictionary);
+
+  columns.push_back(9);
+  ParquetFileReader::MemoryUsage memory_usage2
+      = reader_->EstimateMemoryUsage(false, columns, batch_size);
+  reader_->DebugPrint(ss, columns, batch_size, true);
+  ASSERT_GT(allocator_.MaxMemory(), memory_usage2.memory);
+  ASSERT_TRUE(memory_usage2.has_dictionary);
+  ASSERT_GT(memory_usage2.memory, memory_usage1.memory);
+
+  columns.clear();
+  ParquetFileReader::MemoryUsage memory_usage_all
+      = reader_->EstimateMemoryUsage(false, columns, batch_size);
+  reader_->DebugPrint(ss, columns, batch_size, true);
+  ASSERT_GT(allocator_.MaxMemory(), memory_usage_all.memory);
+  ASSERT_TRUE(memory_usage_all.has_dictionary);
+  ASSERT_GT(memory_usage_all.memory, memory_usage2.memory);
```

Comment:
Agree. I've tested memory estimation on all different combinations of (column type, encoding, compression) offline on a bunch of pre-generated Parquet file, so I'm quite confident in this functionality. But it would be nice to automate and incorporate these tests via faux structures. Do you think this should be done within this patch, or it's big enough to have a separate jira?